### PR TITLE
Raster band and Enum multiple parameters

### PR DESCRIPTION
**Pull request by gavg712**
Please take a look to this two parameters proposal. 

1. First one is Raster band as parameter.

image

2. Second is a enum multiple (literal / index)  

image

I hope, both will be useful in several ways.

Cheers

### CODE REVIEW

File: processing_r/processing/algorithm.py [hunk 1]
```diff
@@ -655,6 +656,9 @@ def build_import_commands(self,  # pylint: disable=too-many-locals,too-many-bran
             if isinstance(param, QgsProcessingParameterRasterLayer):
                 rl = self.parameterAsRasterLayer(parameters, param.name(), context)
                 commands.append(self.build_raster_layer_import_command(param.name(), rl))
+            elif isinstance(param, QgsProcessingParameterBand):
+                value = self.parameterAsDouble(parameters, param.name(), context)
```
Comment (nyalldawson) [inline processing_r/processing/algorithm.py:660]: Shouldn't this be parameterAsInt ? I can't see any case where you'd want a non-integer band number here...

**Review by nyalldawson — COMMENTED** [review]
Looking great, many thanks!

Don't forget to add documentation to https://github.com/north-road/qgis-processing-r/blob/master/website/pages/script-syntax.md also

File: processing_r/processing/algorithm.py (hunk 1 above)
Comment (gavg712) [reply processing_r/processing/algorithm.py:660]: Oh yes, but in R it is not a problem. However, it is better to change it.

Comment (gavg712) [reply processing_r/processing/algorithm.py]: Done

**Comment by gavg712** [comment]
Done, Added documentation for both parameters

File: website/pages/script-syntax.md [hunk 2]
```diff
@@ -50,6 +50,10 @@ So the inputs can look like this:
 
 `##X=Field Layer` specifies that variable `X` will be field name taken from `Layer`.
 
+`##Raster_Layer=vector` specifies that there will be variable `Layer` that will be a raster.
```
Comment (nyalldawson) [inline website/pages/script-syntax.md:53]:
```suggestion
`##Raster_Layer=raster` specifies that there will be variable `Layer` that will be a raster.
```

File: website/pages/script-syntax.md [hunk 3]
```diff
@@ -50,6 +50,10 @@ So the inputs can look like this:
...
+
+`##X=Band Layer` specifies that variable `X` will be raster band index taken from `Raster_Layer`.
```
Comment (nyalldawson) [inline website/pages/script-syntax.md:55]:
```suggestion
`##X=Band Raster_Layer` specifies that variable `X` will be raster band index taken from `Raster_Layer`.
```

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**